airflow/executors/kubernetes_executor.py

```diff
@@ -205,7 +205,7 @@ def process_status(
             self.watcher_queue.put((pod_id, namespace, State.FAILED, annotations, resource_version))
         elif status == 'Succeeded':
             self.log.info('Event: %s Succeeded', pod_id)
-            self.watcher_queue.put((pod_id, namespace, None, annotations, resource_version))
+            self.watcher_queue.put((pod_id, namespace, State.SUCCESS, annotations, resource_version))
```

Comment:
This has been like this for a long time --- @dimberman Am I missing something here?

Reply:
This was changed 3 years back in https://github.com/apache/airflow/pull/3301

cc @dimberman any ideas

Reply:
@ngaranko @kaxil this doesn't need to be set in the executor. Successes are set by the worker when the task completes. Maybe we should add a comment explaining this?